pg/pg.go

```diff
@@ -28,6 +30,54 @@ func NewPG() (*PG, error) {
 	}
 
 	return &PG{
-		Conn: db,
+		Conn:     db,
+		QueryMap: make(map[string]string, 0),
 	}, nil
 }
+
+// Parses a given base dirPath and stores all query files into
+// PG.QueryMap with the format: map[absoluteFilePath]queryFileContents.
+// example: queries/getVendor = "SELECT * FROM vendors WHERE name = $1;"
+//
+// Works recursively through the tree. a nested file may look like:
+// queries/tickets/delete = "DELETE FROM tickets WHERE uuid = $1;"
+func (pg *PG) LoadQueryMap(dirPath string) error {
+	dirs := []string{dirPath}
+	for len(dirs) > 0 {
+		dir := pop(&dirs)
+		files, err := os.ReadDir(dir)
+		if err != nil {
+			return err
+		}
+
+		for _, file := range files {
+			path := fmt.Sprintf("%s/%s", dir, file.Name())
+
+			if file.IsDir() {
+				dirs = append(dirs, path)
+				continue
+			}
+
+			data, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			extension := strings.Split(file.Name(), ".")
+			// Ensure file type (we may want to remove this)
+			if extension[len(extension)-1] != "sql" {
+				continue
+			}
+
+			pg.QueryMap[dir+"/"+extension[0]] = string(data)
+		}
+	}
+	return nil
+}
+
+func pop(arr *[]string) string {
```

Comment:
could move this to utils and make it take any type i guess since go doesn't have a native `pop`